src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs

```diff
@@ -48,14 +48,14 @@ public static partial class JsonSerializer
                         break;
                 }
 
-                if (flushThreshold >= 0 && writer.BytesPending > flushThreshold)
+                if (finishedSerializing && writer.CurrentDepth == 0)
                 {
-                    return false;
```

Comment:
I see - it returned false here to request more data, but we are done serializing. That causes an extra loop which will try to re-serialize the current value.

Reply:
yep, @steveharter can you try to run tailored test also on you master, only to be sure that it's ok, it should fail(like on my machine) without fix.